### PR TITLE
DTS: i.mx: Convert driver to new DT_INST/DT_NODELABEL macros

### PR DESCRIPTION
**Pull request by galak**

### CODE REVIEW

File: boards/arm/colibri_imx7d_m4/pinmux.c [hunk 1]
```diff
@@ -46,9 +46,9 @@ static int colibri_imx7d_m4_pinmux_init(struct device *dev)
 	/* Select TX_PAD for RX data (DTE mode...) */
 	IOMUXC_UART2_RX_DATA_SELECT_INPUT =
 		IOMUXC_UART2_RX_DATA_SELECT_INPUT_DAISY(3);
-#endif  /* CONFIG_UART_IMX_UART_2 */
+#endif
 
-#ifdef CONFIG_I2C_1
+#if DT_HAS_NODE(DT_NODELABEL(i2c))
```
Comment (MaureenHelm) [inline boards/arm/colibri_imx7d_m4/pinmux.c:51]: this should be i2c1

File: boards/arm/colibri_imx7d_m4/pinmux.c [hunk 2]
```diff
@@ -70,9 +70,9 @@ static int colibri_imx7d_m4_pinmux_init(struct device *dev)
 		IOMUXC_SW_PAD_CTL_PAD_I2C1_SDA_PS(3)    |
 		IOMUXC_SW_PAD_CTL_PAD_I2C1_SDA_DSE(0)   |
 		IOMUXC_SW_PAD_CTL_PAD_I2C1_SDA_HYS_MASK;
-#endif  /* CONFIG_I2C_1 */
+#endif
 
-#ifdef CONFIG_I2C_2
+#if DT_HAS_NODE(DT_NODELABEL(i2c))
```
Comment (MaureenHelm) [inline boards/arm/colibri_imx7d_m4/pinmux.c:75]: i2c2

File: boards/arm/colibri_imx7d_m4/pinmux.c [hunk 3]
```diff
@@ -94,9 +94,9 @@ static int colibri_imx7d_m4_pinmux_init(struct device *dev)
 		IOMUXC_SW_PAD_CTL_PAD_I2C2_SDA_PS(3)    |
 		IOMUXC_SW_PAD_CTL_PAD_I2C2_SDA_DSE(0)   |
 		IOMUXC_SW_PAD_CTL_PAD_I2C2_SDA_HYS_MASK;
-#endif  /* CONFIG_I2C_2 */
+#endif
 
-#ifdef CONFIG_I2C_3
+#if DT_HAS_NODE(DT_NODELABEL(i2c))
```
Comment (MaureenHelm) [inline boards/arm/colibri_imx7d_m4/pinmux.c:99]: i2c3

File: boards/arm/colibri_imx7d_m4/pinmux.c [hunk 4]
```diff
@@ -118,9 +118,9 @@ static int colibri_imx7d_m4_pinmux_init(struct device *dev)
 		IOMUXC_SW_PAD_CTL_PAD_I2C3_SDA_PS(3)    |
 		IOMUXC_SW_PAD_CTL_PAD_I2C3_SDA_DSE(0)   |
 		IOMUXC_SW_PAD_CTL_PAD_I2C3_SDA_HYS_MASK;
-#endif  /* CONFIG_I2C_3 */
+#endif
 
-#ifdef CONFIG_I2C_4
+#if DT_HAS_NODE(DT_NODELABEL(i2c))
```
Comment (MaureenHelm) [inline boards/arm/colibri_imx7d_m4/pinmux.c:123]: i2c4

File: boards/arm/pico_pi_m4/pinmux.c [hunk 5]
```diff
@@ -78,10 +78,10 @@ static int pico_pi_m4_pinmux_init(struct device *dev)
 
 	IOMUXC_UART6_RX_DATA_SELECT_INPUT =
 		IOMUXC_UART6_RX_DATA_SELECT_INPUT_DAISY(0);
- #endif  /* CONFIG_UART_IMX_UART_6 */
+ #endif
 
 
- #ifdef CONFIG_I2C_1
+ #if DT_HAS_NODE(DT_NODELABEL(i2c))
```
Comment (MaureenHelm) [inline boards/arm/pico_pi_m4/pinmux.c:84]: i2c1

File: boards/arm/pico_pi_m4/pinmux.c [hunk 6]
```diff
@@ -103,10 +103,10 @@ static int pico_pi_m4_pinmux_init(struct device *dev)
 		IOMUXC_SW_PAD_CTL_PAD_UART1_TX_DATA_PS(3)    |
 		IOMUXC_SW_PAD_CTL_PAD_UART1_TX_DATA_DSE(0)   |
 		IOMUXC_SW_PAD_CTL_PAD_UART1_TX_DATA_HYS_MASK;
- #endif  /* CONFIG_I2C_1 */
+ #endif
 
 
- #ifdef CONFIG_I2C_2
+ #if DT_HAS_NODE(DT_NODELABEL(i2c))
```
Comment (MaureenHelm) [inline boards/arm/pico_pi_m4/pinmux.c:109]: i2c2

File: boards/arm/pico_pi_m4/pinmux.c [hunk 7]
```diff
@@ -128,10 +128,10 @@ static int pico_pi_m4_pinmux_init(struct device *dev)
 		IOMUXC_SW_PAD_CTL_PAD_I2C2_SDA_PS(3)    |
 		IOMUXC_SW_PAD_CTL_PAD_I2C2_SDA_DSE(0)   |
 		IOMUXC_SW_PAD_CTL_PAD_I2C2_SDA_HYS_
- #endif  /* CONFIG_I2C_2 */
+ #endif
 
 
- #ifdef CONFIG_I2C_3
+ #if DT_HAS_NODE(DT_NODELABEL(i2c))
```
Comment (MaureenHelm) [inline boards/arm/pico_pi_m4/pinmux.c:134]: i2c3

File: boards/arm/pico_pi_m4/pinmux.c [hunk 8]
```diff
@@ -153,10 +153,10 @@ static int pico_pi_m4_pinmux_init(struct device *dev)
 		IOMUXC_SW_PAD_CTL_PAD_I2C3_SDA_PS(3)    |
 		IOMUXC_SW_PAD_CTL_PAD_I2C3_SDA_DSE(0)   |
 		IOMUXC_SW_PAD_CTL_PAD_I2C3_SDA_HYS_MASK;
- #endif  /* CONFIG_I2C_3 */
+ #endif
 
 
- #ifdef CONFIG_I2C_4
+ #if DT_HAS_NODE(DT_NODELABEL(i2c))
```
Comment (MaureenHelm) [inline boards/arm/pico_pi_m4/pinmux.c:159]: i2c4

File: boards/arm/warp7_m4/pinmux.c [hunk 9]
```diff
@@ -67,9 +67,9 @@ static int warp7_m4_pinmux_init(struct device *dev)
 	/* Select ECSPI1_SCLK_ALT1 for RX data */
 	IOMUXC_UART6_RX_DATA_SELECT_INPUT =
 		IOMUXC_UART6_RX_DATA_SELECT_INPUT_DAISY(2);
-#endif  /* CONFIG_UART_IMX_UART_6 */
+#endif
 
-#ifdef CONFIG_I2C_1
+#if DT_HAS_NODE(DT_NODELABEL(i2c))
```
Comment (MaureenHelm) [inline boards/arm/warp7_m4/pinmux.c:72]: i2c1

File: boards/arm/warp7_m4/pinmux.c [hunk 10]
```diff
@@ -91,9 +91,9 @@ static int warp7_m4_pinmux_init(struct device *dev)
 		IOMUXC_SW_PAD_CTL_PAD_I2C1_SDA_PS(3)    |
 		IOMUXC_SW_PAD_CTL_PAD_I2C1_SDA_DSE(0)   |
 		IOMUXC_SW_PAD_CTL_PAD_I2C1_SDA_HYS_MASK;
-#endif  /* CONFIG_I2C_1 */
+#endif
 
-#ifdef CONFIG_I2C_2
+#if DT_HAS_NODE(DT_NODELABEL(i2c))
```
Comment (MaureenHelm) [inline boards/arm/warp7_m4/pinmux.c:96]: i2c2

File: boards/arm/warp7_m4/pinmux.c [hunk 11]
```diff
@@ -115,9 +115,9 @@ static int warp7_m4_pinmux_init(struct device *dev)
 		IOMUXC_SW_PAD_CTL_PAD_I2C2_SDA_PS(3)    |
 		IOMUXC_SW_PAD_CTL_PAD_I2C2_SDA_DSE(0)   |
 		IOMUXC_SW_PAD_CTL_PAD_I2C2_SDA_HYS_MASK;
-#endif  /* CONFIG_I2C_2 */
+#endif
 
-#ifdef CONFIG_I2C_3
+#if DT_HAS_NODE(DT_NODELABEL(i2c))
```
Comment (MaureenHelm) [inline boards/arm/warp7_m4/pinmux.c:120]: i2c3

File: boards/arm/warp7_m4/pinmux.c [hunk 12]
```diff
@@ -139,9 +139,9 @@ static int warp7_m4_pinmux_init(struct device *dev)
 		IOMUXC_SW_PAD_CTL_PAD_I2C3_SDA_PS(3)    |
 		IOMUXC_SW_PAD_CTL_PAD_I2C3_SDA_DSE(0)   |
 		IOMUXC_SW_PAD_CTL_PAD_I2C3_SDA_HYS_MASK;
-#endif  /* CONFIG_I2C_3 */
+#endif
 
-#ifdef CONFIG_I2C_4
+#if DT_HAS_NODE(DT_NODELABEL(i2c))
```
Comment (MaureenHelm) [inline boards/arm/warp7_m4/pinmux.c:144]: i2c4

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: boards/arm/warp7_m4/pinmux.c (hunk 12 above)
Comment (galak) [reply boards/arm/warp7_m4/pinmux.c:144]: fixed

File: boards/arm/warp7_m4/pinmux.c (hunk 11 above)
Comment (galak) [reply boards/arm/warp7_m4/pinmux.c:120]: fixed

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**